### PR TITLE
Add com.github.AmatCoder.mednaffe

### PR DESCRIPTION
**Pull request by AmatCoder**
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  *  "--device=all" for controllers.
  *  "--share=network" for netplay.
  *  "--filesystem=home" because mednafen allows you to set the paths to directories for save states, screenshots, etc.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
  *  I am the developer of Mednaffe. Mednaffe is a front-end (GUI) for [mednafen](https://mednafen.github.io/) which is a argument (command-line)-driven multi-system emulator. So this flatpak must to include the emulator. I contacted mednafen developers here: [https://forum.fobby.net/index.php?t=msg&th=1999&start=0&](https://forum.fobby.net/index.php?t=msg&th=1999&start=0&)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  *  Patch submitted [here](https://github.com/AmatCoder/mednaffe/commit/c30df2e51c95ef4172e4ddfb6e285b3383b6bd35).

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by AmatCoder** [comment]
bot, build com.github.mednaffe

File: com.github.mednaffe.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.github.mednaffe</id>
```
Comment (hfiguiere) [inline com.github.mednaffe.appdata.xml:3]: The app-id should be com.github.amatcoder.mednaffe

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.mednaffe.appdata.xml (hunk 1 above)
Comment (AmatCoder) [reply com.github.mednaffe.appdata.xml:3]: Hmm, but id from application is com.github.mednaffe:
https://github.com/AmatCoder/mednaffe/blob/master/src/mednaffe.c#L48

Do I need to patch it?

File: com.github.mednaffe.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                    "type": "archive",
+                    "url": "https://mednafen.github.io/releases/files/mednafen-1.26.1.tar.xz",
+                    "sha256": "842907c25c4292c9ba497c9cb9229c7d10e04e22cb4740d154ab690e6587fdf4"
+                }
+            ]
+        },
+        {
+            "name": "mednaffe",
+            "buildsystem": "autotools",
+            "sources": [
```
Comment (hfiguiere) [inline com.github.mednaffe.json:33]: you meed to install the appstream file.

Comment (AmatCoder) [reply com.github.mednaffe.json:36]: Opps, it is true.

File: com.github.mednaffe.appdata.xml (hunk 1 above)
Comment (AmatCoder) [reply com.github.mednaffe.appdata.xml]: Yes, I need patch it. 
If not it give me errors with DBus.

**Comment by AmatCoder** [comment]
bot, build com.github.mednaffe

File: com.github.mednaffe.appdata.xml (hunk 1 above)
Comment (hfiguiere) [reply com.github.mednaffe.appdata.xml]: yes. The namespace on github is per user/organisation.

File: com.github.mednaffe.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.github.AmatCoder.mednaffe</id>
```
Comment (hfiguiere) [inline com.github.mednaffe.appdata.xml:3]: The files need to be renamed too.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.mednaffe.appdata.xml (hunk 3 above)
Comment (AmatCoder) [reply com.github.mednaffe.appdata.xml]: Ah, thanks.

**Comment by AmatCoder** [comment]
bot, build com.github.mednaffe

**Comment by hfiguiere** [comment]
bot, build com.github.AmatCoder.mednaffe

**Comment by hfiguiere** [comment]
AppData fail to validate:

`• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]`

**Comment by AmatCoder** [comment]
Oh, this will be difficult.
Mednafen can run a wide variety of games. I don't know how to handle this...

**Comment by hfiguiere** [comment]
Not sure how other emulators cover this.

**Comment by hfiguiere** [comment]
bot, build com.github.AmatCoder.mednaffe

**Comment by AmatCoder** [comment]
All emulators seem to disregard the content rating so I will do too.

**Comment by AmatCoder** [comment]
Finally!  :)

@hfiguiere: Thanks very much.

**Comment by barthalion** [comment]
> So this flatpak must to include the emulator.

Technically speaking, we're concerned only about your involvement into the main project (i.e. frontend), not every dependency possible, but it's up to you.

File: com.github.AmatCoder.mednaffe.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.github.AmatCoder.mednaffe.appdata.xml:1]: Ideally you should maintain it in your own repo.

Comment (AmatCoder) [reply com.github.AmatCoder.mednaffe.appdata.xml:1]: Yes, but I would like to give permission to edit it to mednafen author.

**Comment by AmatCoder** [comment]
This can be merged.

**Pull request closed without merging**